pkg/utils/resources/suite_test.go

```diff
@@ -0,0 +1,409 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"sigs.k8s.io/karpenter/pkg/test"
+
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestResources(t *testing.T) {
```

Comment:
Can we make the limits and resources match in all of these tests so that it's a little less confusing to reason about? Then, you could just create one additional test that validated the behavior with different limits/requests on each. Reading through, it's hard to reason about some of the test cases when the resource requests are fulfilling the description of the test, but the limits aren't tracking with that